### ISSUE TITLE
Translation example dev/test mismatch

### ISSUE THREAD

**Issue by ZhaofengWu** [pytorch/fairseq#3128]
## ❓ Questions and Help

In the translation example (https://github.com/pytorch/fairseq/blob/master/examples/translation/README.md#iwslt14-german-to-english-transformer), during `fairseq-train`, the dev set uses arguments `--eval-bleu-args '{"beam": 5, "max_len_a": 1.2, "max_len_b": 10}' --eval-bleu-detok moses --eval-bleu-remove-bpe`, but during `fairseq-generate`, the test set uses `--beam 5 --remove-bpe`, without (a) `max_len_*` and (b) detokenization. Why is this mismatch?
